### DISCUSSION TITLE
Clip skip?

### DISCUSSION THREAD

**Discussion by MIC132** [azuritecoin/OnnxDiffusersUI#71] [Q&A]
Does this tool have the ability to set clip skip, or is it planned?

It's necessary for certain models.

**Comment by azuritecoin** [comment]
You mean clip skip the last iteration right? This tool doesn't do that right now.

**Reply by MIC132** [reply]
Last iteration?
Honestly I always see it just called "clip skip <value>", and normal is 1, but some models need 2.
If I understand correctly it's something about taking tokens earlier from the CLIP network.

**Comment by uchuusen** [comment]
Clip skip can be achieved by editing the config.json file in the text_encoder directory of the diffusers model, before converting to ONNX. You should find the line:
`"num_hidden_layers": 12,`
Just change the 12 to 11, then convert to ONNX and it should work. The resulting ONNX model should be the same as setting clip skip to 2 in A1111.
See https://github.com/huggingface/diffusers/issues/1721 for the tech details.
